src/components/rule-builder/rule-builder.component.tsx

```diff
@@ -0,0 +1,1028 @@
+    isNewRule,
+    index,
+    deleteCondition,
+    addCondition,
+    launchDeleteConditionalLogicModal,
+    answers,
+    handleConditionChange,
+    conditions,
+    addNewConditionalLogic,
+  }: RuleConditionProps) => {
+    const { t } = useTranslation();
+    const answer = answers.filter((answer) => answer !== undefined);
+    const [isConditionValueVisible, setIsConditionValueVisible] = useState<boolean>(
+      Boolean(conditions[index]?.targetValue),
+    );
+    const handleSelectCondition = (selectedCondition: string) => {
+      setIsConditionValueVisible(!['Is Empty', 'Not Empty'].includes(selectedCondition));
+    };
+    const [inputValue, setInputValue] = useState(conditions[index]?.targetValue || '');
+    const handleValueChange = (selectedItem: string) => {
+      handleConditionChange(fieldId, 'targetValue', selectedItem, index);
+      setInputValue(selectedItem);
+    };
+
+    return (
+      <div className={styles.ruleSetContainer}>
+        <div className={styles.sectionContainer}>
+          {isNewCondition ? (
+            <Dropdown
+              id={`logicalOperator-${index}`}
+              className={styles.logicalOperator}
+              initialSelectedItem={conditions[index]?.logicalOperator || 'and'}
+              defaultSelectedItem="and"
+              items={['and', 'or']}
+              onChange={({ selectedItem }: { selectedItem: string }) => {
+                handleConditionChange(fieldId, `logicalOperator`, selectedItem, index);
+              }}
+              size={isTablet ? 'lg' : 'sm'}
+            />
+          ) : (
+            <div className={styles.ruleDescriptor} id="when-rule-descriptor">
+              <span className={styles.icon}>
+                <FlowConnection />
+              </span>
+              <p className={styles.label}>{t('when', 'When')}</p>
+            </div>
+          )}
+          <Dropdown
+            id={`targetField-${index}`}
+            className={styles.targetField}
+            initialSelectedItem={
+              questions?.find((question) => question.id === conditions[index]?.[`targetField`]) || {
+                label: 'Choose a field',
+              }
+            }
+            items={questions}
+            itemToString={(item: Question) => (item ? item.label : '')}
+            onChange={({ selectedItem }: { selectedItem: Question }) =>
+              handleConditionChange(fieldId, `targetField`, selectedItem.id, index)
+            }
+            size={isTablet ? 'lg' : 'sm'}
+          />
+          <Dropdown
+            id={`targetCondition-${index}`}
+            aria-label="target-condition"
+            className={styles.targetCondition}
+            selectedItem={conditions[index]?.[`targetCondition`] || 'Select Condition'}
+            items={[
+              'Is Empty',
+              'Not Empty',
+              'Greater than or equal to',
+              'Less than or equal to',
+              'Equals',
+              'not Equals',
```

Comment:
We could put this in a collection somewhere so they're sufficiently translated and annotated with types.